Domain/EventSourcedAggregate{T}.cs

```diff
@@ -64,11 +64,12 @@ protected void BuildUpStateFromEventHistory()
         /// <summary>
         ///     Records an event, updating the aggregate's state and adding the event to PendingEvents.
         /// </summary>
-        /// <remarks>It is not necessary to specify the AggregateId or SequenceNumber properties on the recorded event. The <see cref="EventSourcedAggregate" /> class handles this.</remarks>
-        protected void RecordEvent(Event<T> e)
-        {
-            AddPendingEvent(e);
-            e.Update((T) this);
+        /// <remarks>It is not necessary to specify the AggregateId or SequenceNumber properties on the recorded event. The <see cref="EventSourcedAggregate" /> class handles this.</remarks>
+        protected Event<T> RecordEvent(Event<T> e)
+        {
+            AddPendingEvent(e);
+            e.Update((T)this);

```

Comment:
one thing that occurred to me here is that the update method on the event can actually allow callers to modify the event properties... i'm not sure why anyone would ever do this but i would likely cause a break if they mucked with the etag.

this issue exists regardless of returning the event here or not though.